src/routes/Library.svelte

```diff
@@ -0,0 +1,179 @@
+<script lang="ts">
+	import { getGames, deleteGame, runGame } from '../scripts/Library';
+	import { getContext } from 'svelte';
+	import NewGame from './modals/NewGame.svelte';
+	import '../styles/Library.scss';
+	import { t } from '../locale/i18n';
+	import { convertFileSrc } from '@tauri-apps/api/tauri';
+
+	// Gets the open function from simple-modal context
+	const { open }: any = getContext('simple-modal');
+
+	// When the modal is closed re-run the function getGames
+	const showNewModal = () =>
+		open(
+			NewGame,
+			{},
+			{},
+			{
+				onClose: () => {
+					games = getGames();
+				},
+			}
+		);
+	// show modal to edit games using prepolulated values
+	const showEditModal = (game: any) =>
+		open(
+			NewGame,
+			{
+				id: game.id,
+				title: game.name,
+				description: game.description,
+				imagePath: game.image,
+				executablePath: game.exe_path,
+				operationToPerform: 'Edit',
+			},
+			{},
+			{
+				onClose: () => {
+					games = getGames();
+				},
+			}
+		);
+
+	var query: string;
+
+	let games: any = getGames();
+	function operationHandler(operation: () => void) {
+		operation();
+		games = getGames();
+	}
+</script>
+
+<main class="container">
+	<div class="main">
+		<div class="top">
+			<h1 style="display:inline-block;">{$t('libraryText')}</h1>
+			<!-- Creates a modal when the button is clicked -->
+			<input
+				type="text"
+				placeholder="Search"
+				class="search-bar"
+				bind:value="{query}"
+			/>
+			<button on:click="{() => showNewModal()}"
+				>{$t('library.add')}</button
+			>
+		</div>
+
+		<!-- Awaits for games to be resolved -->
+		<!-- After that, loop over every object in that array -->
+		<!-- And add those results to a div -->
+		<!-- svelte-ignore empty-block -->
+		{#await games then data}
+			{#each data as game}
+				{#if !query}
```

Comment:
This thing can be simplified and the code can be reduced to half if we implement a function that returns a new array of filtered results matching the query, or returning the whole array if query is empty.

Use this demo function:
```typescript
function getFilteredGames(games: array[], gameToSearch?: string){
    if(typeof gameToSearch !== 'undefined'){
        return games.filter(game => {
            return game.title.toLowerCase().includes(gameToSearch.toLowerCase());
        });
    } else {
        return games;
    }
}
```

The above function is not tested by me. Please do necessary checks.